### PR TITLE
Provide save option in GIS tool; replace es case_json usage

### PR DESCRIPTION
**Pull request by sravfeyn**
## Product Description

This does two independent things
1. Adds the standard 'Save' button to the GIS tool page
2. Replaces use of ES `case_json` attribute with direct case-property lookups. See relevant discussion [here](https://dimagi.slack.com/archives/C02QG63MU/p1689687231739319) which otherwise results in bugs on domains that contain pre couch-to-sql-migrated cases.

## Technical Summary

3rd item from [this spec](https://docs.google.com/document/d/1o7O8pnBfz8Eah_x9amAfnxKW_tCuOI8Fe9aPazk5OGY/edit#heading=h.iu4xk6min08b)

## Feature Flag

`GEOSPATIAL`

### Safety story

Very simple change that has minimal touchpoints.

### Automated test coverage

Tested locally

### QA Plan

NA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by Charl1996 — APPROVED** [review]

**Comment by Charl1996** [comment]
@sravfeyn 
The import in `corehq/apps/geospatial/tests/test_report.py` should just be fixed (which is causing the failing test on the PR)

File: corehq/apps/geospatial/reports.py [hunk 1]
```diff
@@ -53,6 +40,24 @@ def template_context(self):
     def report_context(self):
         cases = []
         invalid_geo_cases_count = 0
+
+        def _get_geo_location(case):
```
Comment (snopoke) [inline corehq/apps/geospatial/reports.py:44]: you could use the [`GeoPoint`](https://github.com/dimagi/commcare-hq/blob/663ba850feae022a552202e814a579782358fe04/corehq/ex-submodules/couchforms/geopoint.py#L8) class to parse the property value

File: corehq/apps/geospatial/tests/test_report.py [hunk 2]
```diff
@@ -1,13 +0,0 @@
-from django.test import TestCase
```
Comment (sravfeyn) [inline corehq/apps/geospatial/tests/test_report.py:1]: I removed this test since _get_geo_location is refactored to just call `Geopoint.from_string`

**Comment by sravfeyn** [comment]
Bumping if this is g2g

**Review by Charl1996 — APPROVED** [review]
Looks good to me!

**Pull request merged**